util/marketplace.py

```diff
@@ -0,0 +1,170 @@
+import json
+import logging
+import requests
+
+from data.billing import RH_SKU_STRIPE_IDS, RH_SKUS, get_plan
+from datetime import datetime
+
+from data.model import entitlements
+
+logger = logging.getLogger(__name__)
+
+REQUEST_TIMEOUT = 60
+
+MARKETPLACE_FILE = "/conf/stack/auth/quay-marketplace-api-stage.crt"
+MARKETPLACE_SECRET = "/conf/stack/auth/quay-marketplace-api-stage.key"
+
+
+class RHUserAPI:
+    def __init__(self, app_config):
+        self.cert = (MARKETPLACE_FILE, MARKETPLACE_SECRET)
+        self.user_endpoint = app_config.get("ENTITLEMENT_RECONCILIATION_USER_ENDPOINT")
+
+    def get_account_number(self, user):
+        email = user.email
+        account_number = entitlements.get_ebs_account_number(user.id)
+        if account_number is None:
+            account_number = self.lookup_customer_id(email)
+            if account_number:
+                # store in database for next lookup
+                entitlements.save_ebs_account_number(user, account_number)
+        return account_number
+
+    def lookup_customer_id(self, email):
+        """
+        Send request to internal api for customer id (ebs acc number)
+        """
+        request_body_dict = {
+            "by": {"emailStartsWith": email},
+            "include": {
+                "accountRelationships": [
+                    {
+                        "allOf": [
+                            "primary_email",
+                            "is_supportable",
+                            "account_details",
+                        ],
+                        "by": {"active": True},
+                    }
+                ]
+            },
+        }
+
+        request_url = f"{self.user_endpoint}/v2/findUsers"
+
+        r = requests.request(
+            method="post",
+            url=request_url,
+            cert=self.cert,
+            json=request_body_dict,
+            verify=True,
+            timeout=REQUEST_TIMEOUT,
+        )
+
+        info = json.loads(r.content)
+        if not info:
+            return None
+        CustomerId = info[0]["accountRelationships"][0]["account"]["ebsAccountNumber"]
```

Comment:
Any particular reason this is CamelCased?

Reply:
No reason, I'm going to rename this to account_number for consistency